File: Selenium/Project/Activity3.py
#Get the copyright text
#Goal: Print the first copyright text in the footer to the console


# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service = service) as driver:
  
  # Open the browser to the URL
  driver.get("https://alchemy.hguy.co/crm")

  # Read the title of the website and verify the text
  print("Home page title:"+driver.title)
  
  #Get the xpath of  footer
  footer =driver.find_element(By.XPATH,"//div[@class='p_login_bottom']")

  #getText() to Get the first copyright text in the footer by using tagname a
  firstCopyrightText=footer.find_elements(By.TAG_NAME,"a")[0].text
  print("The first copyright text in the footer is:"+firstCopyrightText)

  # Close the browser
  driver.close()

File: Selenium/Project/Activity3.java
/*Get the copyright text
        Goal: Print the first copyright text in the footer to the console*/

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class Activity3 {
    // Declare the WebDriver object
    private WebDriver driver;

    @BeforeMethod
    public void setupMethod() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        //Open browser and navigate to url
        driver.get("https://alchemy.hguy.co/crm");
    }

    @Test
    public void getCopyrightText() {
        //Get the xpath of  footer
        WebElement footer =driver.findElement(By.xpath("//div[@class='p_login_bottom']"));

        //getText() to Get the first copyright text in the footer by using tagname a
        String firstCopyrightText=footer.findElements(By.tagName("a")).get(0).getText();
        System.out.println("The first copyright text in the footer is:"+firstCopyrightText);
    }

    @AfterMethod
    public void afterMethod() {
        //Close the browser
        driver.close();
    }
}
